util.c

```diff
@@ -5008,6 +5008,9 @@ S_mem_log_common(enum mem_log_type mlt, const UV n,
 
     PERL_ARGS_ASSERT_MEM_LOG_COMMON;
 
+    /* Note that 'pmlenv' is not protected from other threads overwriting it by
+     * calling getenv() themselves on platforms where getenv() returns an
+     * internal static pointer. */
     pmlenv = PerlEnv_getenv("PERL_MEM_LOG");
```

Comment:
Why not just call getenv() here?  That's what happens anyway, and doesn't cost every other Perl_mortal_getenv() the (minor) cost of a string comparison.

Reply:
That is a much better solution than mine.  Updated to do what you say.